BrainPortal/lib/scir_lsf.rb

```diff
@@ -28,73 +28,62 @@ class ScirLsf < Scir
 
   class Session < Scir::Session #:nodoc:
 
-    def update_job_info_cache #:nodoc:
-      # FIXME: modify this method based
+    def update_job_info_cache #:nodoc:      
       # on the implementation of bstat.  This method is supposed to
       # list all the statuses of the running tasks, and to keep it in
       # the @job_info_cache array.
-      out, err = bash_this_and_capture_out_err("bstat -f -u #{CBRAIN::Rails_UserName.to_s.bash_escape}")
-      raise "Cannot get output of 'bstat -f' ?!?" if out.blank? && ! err.blank?
+      out, err = bash_this_and_capture_out_err("bjobs -a -noheader")
+      raise "Cannot get output of 'bjobs -a' ?!?" if out.blank? && ! err.blank?
```

Comment:
Can't you filter the jobs that belong to a particular user, as it is done with option -u in qsub? Otherwise it will put a lot of unnecessary stuff in memory when clusters are shared with other users.